### PR TITLE
Unbound: Clarify “Verify CN” option field

### PR DESCRIPTION
**Pull request by da2x**
A small option string change that’ll hopefully make it clearer what the setting is for and what to enter.

### CODE REVIEW

File: src/opnsense/mvc/app/controllers/OPNsense/Unbound/forms/dialogDot.xml [hunk 1]
```diff
@@ -31,8 +31,8 @@
         <label>Verify CN</label>
         <type>text</type>
         <help>
-            Verify if CN in certificate matches this value. Please note that if this field is omitted,
-            Unbound will not perform any certificate verification. Therefore, man-in-the-middle attacks are still possible.
+            Verify TLS certificates against the specified Common Name (e.g. dns.example.com). Please note that if this field is
+            omitted, Unbound will not perform any certificate verification. Therefore, man-in-the-middle attacks are still possible.
```
Comment (fichtner) [inline src/opnsense/mvc/app/controllers/OPNsense/Unbound/forms/dialogDot.xml:35]: Would you mind leaving the second line as is and move the omitted up?

File: src/opnsense/mvc/app/controllers/OPNsense/Unbound/forms/dialogDot.xml [hunk 2]
```diff
@@ -31,8 +31,8 @@
         <label>Verify CN</label>
         <type>text</type>
         <help>
-            Verify if CN in certificate matches this value. Please note that if this field is omitted,
-            Unbound will not perform any certificate verification. Therefore, man-in-the-middle attacks are still possible.
+            Verify TLS certificates against the specified Common Name (e.g. dns.example.com). Please note that if this field is
```
Comment (fichtner) [inline src/opnsense/mvc/app/controllers/OPNsense/Unbound/forms/dialogDot.xml:34]: Might even make sense to clarify which TLS certificate is verified, i.e. where they come from. 😊

File: src/opnsense/mvc/app/controllers/OPNsense/Unbound/forms/dialogDot.xml (hunk 1 above)
Comment (da2x) [reply src/opnsense/mvc/app/controllers/OPNsense/Unbound/forms/dialogDot.xml]: I rewrote it again. I think its clearer now? It's more detailed, at least.

File: src/opnsense/mvc/app/controllers/OPNsense/Unbound/forms/dialogDot.xml [hunk 3]
```diff
@@ -31,8 +31,9 @@
         <label>Verify CN</label>
         <type>text</type>
         <help>
-            Verify if CN in certificate matches this value. Please note that if this field is omitted,
-            Unbound will not perform any certificate verification. Therefore, man-in-the-middle attacks are still possible.
+            The DNS servers' Common Name (e.g. dns.example.com). This field is required to verify the server's TLS certificate.
```
Comment (fichtner) [inline src/opnsense/mvc/app/controllers/OPNsense/Unbound/forms/dialogDot.xml:34]:
```suggestion
            The Common Name of the DNS server (e.g. dns.example.com). This field is required to verify its TLS certificate.
```

File: src/opnsense/mvc/app/controllers/OPNsense/Unbound/forms/dialogDot.xml [hunk 4]
```diff
@@ -31,8 +31,9 @@
         <label>Verify CN</label>
         <type>text</type>
         <help>
-            Verify if CN in certificate matches this value. Please note that if this field is omitted,
-            Unbound will not perform any certificate verification. Therefore, man-in-the-middle attacks are still possible.
+            The Common Name of the DNS server (e.g. dns.example.com). This field is required to verify its TLS certificate.
+            DNS-over-TLS is susceptible to man-in-the-middle attacks unless certificates can be verified.
+            Leave empty to accept self-signed (and potentially fraudulent) certificates.
```
Comment (fichtner) [inline src/opnsense/mvc/app/controllers/OPNsense/Unbound/forms/dialogDot.xml:36]:
```suggestion
            Leave empty to accept self-signed yet also potentially fraudulent certificates.
```

**Comment by fichtner** [comment]
Merged, thanks!

**Pull request merged**